AettirOgLandsvaedi_Lidur1/aettiroglandsvaedi.sql

```diff
@@ -0,0 +1,253 @@
+    WHERE match_priority < 5
+)
+-- Upsertum í töfluna
+INSERT INTO martell.tables_mapping (house_id, location_id)
+SELECT house_id, location_id
+FROM filterum
+WHERE row_num = 1 AND house_row_num = 1
+ON CONFLICT (house_id)
+DO UPDATE
+SET location_id = EXCLUDED.location_id;
+
+
+-- Sýnir einungis niðurstöður fyrir norðrið (e. The North)
+WITH location_house_mapping AS (
+    SELECT
+        l.gid AS location_id, --ýta hér á kóðann til að fá upp réttu statement skipanir í keyrsluhnapp
+        h.id AS house_id,
+        l.name AS location_name,
+        h.name AS house_name,
+        h.region AS house_region,
+        h.seats,
+        h.titles,
+        l.summary AS location_summary,
+        CASE
+            WHEN h.name ILIKE '%' || l.name || '%' AND
+                 h.name ~* ('\m' || l.name || '\M') THEN 1
+            WHEN EXISTS (
+                SELECT 1
+                FROM UNNEST(h.seats) AS seat
+                WHERE seat ILIKE '%' || l.name || '%'
+                AND seat ~* ('\m' || l.name || '\M')
+            ) THEN 2
+            WHEN EXISTS (
+                SELECT 1
+                FROM UNNEST(h.titles) AS title
+                WHERE title ILIKE '%' || l.name || '%'
+                AND title ~* ('\m' || l.name || '\M')
+            ) THEN 3
+            WHEN l.summary ILIKE '%' || h.name || '%' AND
+                 l.summary ~* ('\m' || h.name || '\M') THEN 4
+            ELSE 5
+        END AS match_priority,
+        COALESCE(
+            (SELECT seat FROM UNNEST(h.seats) AS seat
+             WHERE seat ILIKE '%' || l.name || '%'
+             AND seat ~* ('\m' || l.name || '\M') LIMIT 1),
+            (SELECT title FROM UNNEST(h.titles) AS title
+             WHERE title ILIKE '%' || l.name || '%'
+             AND title ~* ('\m' || l.name || '\M') LIMIT 1),
+            h.name
+        ) AS matched_detail
+    FROM
+        atlas.locations l
+    LEFT JOIN
+        got.houses h
+    ON
+        (h.name ILIKE '%' || l.name || '%' AND h.name ~* ('\m' || l.name || '\M'))
+        OR EXISTS (
+            SELECT 1
+            FROM UNNEST(h.seats) AS seat
+            WHERE seat ILIKE '%' || l.name || '%'
+            AND seat ~* ('\m' || l.name || '\M')
+        )
+        OR EXISTS (
+            SELECT 1
+            FROM UNNEST(h.titles) AS title
+            WHERE title ILIKE '%' || l.name || '%'
+            AND title ~* ('\m' || l.name || '\M')
+        )
+        OR (l.summary ILIKE '%' || h.name || '%' AND l.summary ~* ('\m' || h.name || '\M'))
+),
+filterum AS (
+    SELECT
+        location_id,
+        house_id,
+        ROW_NUMBER() OVER (PARTITION BY location_id ORDER BY match_priority, house_name) AS row_num,
+        ROW_NUMBER() OVER (PARTITION BY house_id ORDER BY match_priority, location_name) AS house_row_num
+    FROM
+        location_house_mapping
+    WHERE match_priority < 5
+)
+SELECT *
+FROM location_house_mapping
+WHERE match_priority < 4
+AND house_region = 'The North';
```

Comment:
Af hverju að reikna þetta allt aftur í CTE þegar þið hafið nú þegar leyst þetta með upsertinu fyrir `martell.tables_mapping`

```suggestion
SELECT houses.name as house_name, locations.name as location_name
FROM martell.tables_mapping inner join got.houses on tables_mapping.house_id = houses.id
inner join atlas.locations on tables_mapping.location_id = locations.gid
where region = 'The North';
```